include/wx/textctrl.h

```diff
@@ -127,19 +127,42 @@ enum wxTextCtrlHitTestResult
 // ... the character returned
 
 // ----------------------------------------------------------------------------
-// wxTextCtrl spell checking options
+// wxTextCtrl proof options object
+// Passed to ::EnableProofCheck() to configure the proofing options for
+// this control
 // ----------------------------------------------------------------------------
-enum wxTextCtrlSpellCheckOptions
+class WXDLLIMPEXP_CORE wxTextProofOptions
 {
-    wxTE_SPELL_OPTION_ALL_CHECKS =              0x0000,
-    wxTE_SPELL_OPTION_SPELL =                   0x0001,
-    wxTE_SPELL_OPTION_GRAMMAR =                 0x0002,
-    wxTE_SPELL_OPTION_AUTOCORRECT =             0x0004,
-    wxTE_SPELL_OPTION_NO_CONTEXT_MENU =         0x0008,
-    wxTE_SPELL_OPTION_NO_IGNORE_WORD =          0x0010,
-    wxTE_SPELL_OPTION_NO_PERSONAL_DICTIONARY =  0x0020
+public:
+    wxTextProofOptions(const wxString& lang = wxEmptyString)
+    : m_lang(lang)
+    {
+        m_EnableSpellCheck = true;
+        m_EnableGrammerCheck = false;
+    }
+
+    wxTextProofOptions& SpellCheck(bool enable = true)
+    {
+        m_EnableSpellCheck = enable;
+        return *this;
+    }
+
+    wxTextProofOptions& GrammerCheck(bool enable = true)
+    {
+        m_EnableGrammerCheck = enable;
+        return *this;
+    }
+
+    const wxString& GetLang() const { return m_lang; }
+
+private:
+    wxString m_lang;
+    bool m_EnableSpellCheck;
+    bool m_EnableGrammerCheck;
 };
 
+extern WXDLLIMPEXP_DATA_CORE(const wxTextProofOptions) wxDefaultTextProofOptions;
```

Comment:
Please check that this declaration is correct.

Reply:
On second thoughts, do we even need this? 

I think it would be okay just to pass the default `wxTextProofOptions` constructor as the default parameter, e.g., 

```
    virtual bool EnableProofCheck(bool WXUNUSED(enable) = true,
                                  const wxTextProofOptions& WXUNUSED(options) =
                                  wxTextProofOptions()) wxOVERRIDE;
```

Reply:
Yes, it is correct and yes, I also agree that we don't need it, using `= wxTextProofOptions()` is also perfectly fine.